terraform/dns/README.md

```diff
@@ -0,0 +1,77 @@
+- The Terraform resources can be committed to your VCS for centralizing the source of truth, keeping declarations and state in sync, automated deployments, regression testing, policy enforcement, and inspecting the history of changes.
+- Leverage the full power of DNSimple's functionality and tools, like ALIAS and regional records, to empower your architecture.
+- Take advantage of all of Terraform's benefits with DNS records that are just like any other resource: drift detection, one-command deployment and synchronization, clean and structured deletion, automatic dependency resolution, track and centralize state, and more.
+
+Feel free to use this proof-of-concept as the starting point for your next project, or as inspiration to swap parts out to build something completely different.
+
+## Prerequisites
+
+To run this demo, you will need:
+
+- An authoritative zone on DNSimple, where the DNS resolution is not delegated or disabled. If you purchased or transferred your domain to DNSimple, or you have pointed your domain's nameservers to DNSimple at your registrar, you should be good to go.
+  - If you'd like to registrar and set up a domain also entirely through Terraform, check out our [domains demo](../domains).
+- An API token for your DNSimple account must be generated and stored in your environment variables. See [here](https://support.dnsimple.com/articles/api-access-token/) for more details.
+- An AWS account. You must have AWS credentials on your machine ready to go. See [this guide](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#authentication-and-configuration) for more details.
+- Terraform must be installed. You can download it [here](https://developer.hashicorp.com/terraform/downloads).
+
+> [!WARNING]
+> DNS records will be overwritten on the zone, which may erase any current records and break services running on your domain.
+
+> [!NOTE]
+> This will create resources on your AWS account, which may incur charges. After running this demo, if you don't need it anymore, make sure to delete the resources.
+
+## Running
+
+This demo has been designed such that you only need to follow along the code in [main.tf](./main.tf). Each block has helpful, descriptive comments that are designed to be read like a guide sequentially, and build on top of previous code and comments.
+
+To get started, set up the Terraform project. Only one variable is required to run the demo, which is the domain you want to test with. Replace `mydomain.com` with your domain name.
+
+```bash
+terraform init
+export TF_VAR_domain=mydomain.com
+```
+
+Now, you can run the entire demo all at once, by running:
+
+```bash
+terraform apply
+```
+
+You can also run the demo incrementally, at your pace, by commenting out all code in [main.tf](./main.tf) after the point where you'd like to stop, and then running the above `terraform apply` command. Once you're ready to proceed, uncomment out some more code, and then repeat.
```

Comment:
May be helpful to include a screenshot of what the user will see in their browser. I know you include this at the end of `mian.tf` but someone might not see it.